piet-wgsl/shader/fine.wgsl

```diff
@@ -273,20 +277,23 @@ fn main(
             default: {}
         }
     }
-    let out_ix = global_id.y * (config.width_in_tiles * TILE_WIDTH) + global_id.x * PIXELS_PER_THREAD;
+    let xy_uint = vec2<u32>(xy);
     for (var i = 0u; i < PIXELS_PER_THREAD; i += 1u) {
-        let fg = rgba[i];
-        let a_inv = 1.0 / (fg.a + 1e-6);
-        let rgba_sep = vec4(fg.r * a_inv, fg.g * a_inv, fg.b * a_inv, fg.a);
-        let bytes = pack4x8unorm(rgba_sep);
-        output[out_ix + i] = bytes;
-    }
+        let coords = xy_uint + vec2(i, 0u);
+        if coords.x < config.target_width && coords.y < config.target_height {
+            textureStore(output, vec2<i32>(coords), rgba[i]);
```

Comment:
According to the WGSL spec, coords can be u32, but naga complains. This might be a naga issue.

Reply:
Yeah, I had tried u32 first, but it yelled at me. I'm sure it'll be fixed in time.